app/controllers/application_controller.rb

```diff
@@ -25,9 +26,19 @@ class ApplicationController < ActionController::Base
   private
 
     def require_active_account!
-      account = Account.from_request(request)
+      raise Apartment::TenantNotFound, "No tenant for #{request.host}" unless current_account
+    end
+
+    def set_account_specific_connections!
+      if current_account && current_account.solr_endpoint
+        solr_endpoint = current_account.solr_endpoint
+        ActiveFedora::SolrService.register(solr_endpoint.url, solr_endpoint.connection_options)
+        Blacklight.instance_variable_set(:@default_index, solr_endpoint.connection)
```

Comment:
This will be better after https://github.com/projectblacklight/blacklight/pull/1403 is merged and released.